iceoryx_utils/source/posix_wrapper/shared_memory_object/shared_memory.cpp

```diff
@@ -214,10 +192,11 @@ bool SharedMemory::unlink() noexcept
 
 bool SharedMemory::close() noexcept
 {
-    if (m_isInitialized)
+    if (m_isInitialized && m_handle != -1)
```

Comment:
So we do not have the invariant that when m_isInitialized is true the m_handle is valid? If so, it would not be needed.